driver-reactive-streams/src/test/unit/com/mongodb/reactivestreams/client/internal/AggregatePublisherImplTest.java

```diff
@@ -58,6 +58,7 @@ void shouldBuildTheExpectedOperation() {
 
         AggregateOperation<Document> expectedOperation = new AggregateOperation<>(NAMESPACE, pipeline,
                                                                                   getDefaultCodecRegistry().get(Document.class))
+                .batchSize(Integer.MAX_VALUE)
```

Comment:
Demand is now signalled and the batchSize set via the requested data amount.

Reply:
We expect batch size to be `Integer.MAX_VALUE`, but I do not see us requesting `Integer.MAX_VALUE` (or more), which is what we need to do in order to get `Integer.MAX_VALUE`, according to the `BatchCursorFlux.calculateBatchSize` method. Could you please help me understand where `Integer.MAX_VALUE` comes from?

Reply:
Sure, by using the blocking functions we are requesting data.

So ` Flux.from(publisher).blockFirst();` requests `Long.MAX_VALUE`, which as batchSize is an int32 gets converted to `Integer.MAX_VALUE`